send.c

```diff
@@ -602,14 +602,23 @@ static struct safe_buffer_list *add_ra_options_rdnss(struct safe_buffer_list *sb
 {
 	while (rdnss) {
 		struct nd_opt_rdnss_info_local rdnssinfo;
-
 		if (!cease_adv && !schedule_option_rdnss(dest, iface, rdnss)) {
 			rdnss = rdnss->next;
 			continue;
 		}
 
 		memset(&rdnssinfo, 0, sizeof(rdnssinfo));
 
+		size_t const bytes = sizeof(rdnssinfo) + sizeof(struct in6_addr) * rdnss->AdvRDNSSNumber;
+		// dnsslinfo.nd_opt_rdnssi_len is uint8 count of 8-octet groups; min 3, max 255
+		// too many DNS servers could exceed it
+		// https://datatracker.ietf.org/doc/html/rfc8106#section-5.1
+		if (bytes > (255 * 8)) {
+			flog(LOG_ERR, "RDNSS too long for RA option, must be < 2032 bytes. Skipping option.");
```

Comment:
Same comment as the other PR - shouldn't this be <= 2040 bytes? Or am I not understanding the math?